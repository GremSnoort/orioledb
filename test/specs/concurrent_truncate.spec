setup
{
	CREATE EXTENSION IF NOT EXISTS orioledb;
	CREATE TABLE t (i int) USING orioledb;
    INSERT INTO t VALUES (45), (66), (233);
}

teardown
{
	DROP TABLE t;
}

session "s1"

step "s1_begin" { BEGIN; }
step "s1_truncate" { TRUNCATE t; }
step "s1_insert" { INSERT INTO t VALUES (6), (19), (40); }
step "s1_select" { SELECT * FROM t; }
step "s1_commit" { COMMIT; }

session "s2"

step "s2_begin" { BEGIN; }
step "s2_truncate" { TRUNCATE t; }
step "s2_insert" { INSERT INTO t VALUES (79), (13), (22); }
step "s2_select" { SELECT * FROM t; }
step "s2_commit" { COMMIT; }

permutation "s1_begin"      "s2_begin" 
            "s1_truncate"
                            "s2_truncate"
            "s1_commit"
            "s1_truncate"
                            "s2_truncate" # one non-transactional truncate just to spice things up a bit
                            "s2_commit"
            "s1_insert"
                            "s2_insert"
            "s1_select"
                            "s2_select" 
