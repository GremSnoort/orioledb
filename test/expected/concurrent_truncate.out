Parsed test spec with 2 sessions

starting permutation: s1_begin s2_begin s1_truncate s2_truncate s1_commit s1_truncate s2_truncate s2_commit s1_insert s2_insert s1_select s2_select
step s1_begin: BEGIN;
step s2_begin: BEGIN;
step s1_truncate: TRUNCATE t;
step s2_truncate: TRUNCATE t; <waiting ...>
step s1_commit: COMMIT;
step s2_truncate: <... completed>
step s1_truncate: TRUNCATE t; <waiting ...>
step s2_truncate: TRUNCATE t;
step s2_commit: COMMIT;
step s1_truncate: <... completed>
step s1_insert: INSERT INTO t VALUES (6), (19), (40);
step s2_insert: INSERT INTO t VALUES (79), (13), (22);
step s1_select: SELECT * FROM t;
 i
--
 6
19
40
79
13
22
(6 rows)

step s2_select: SELECT * FROM t;
 i
--
 6
19
40
79
13
22
(6 rows)

